### ISSUE TITLE
Installation error - Numpy and its headers are required to run setup()

### ISSUE THREAD

**Issue by giswqs** [developmentseed/titiler#158]
Both installation methods failed on my computer with a fresh conda env. After explicitly installing numpy, then the installation went smoothly. Does numpy need to be strictly declared dependency for titiler? I found this relevant [issue](https://github.com/mojodna/marblecutter-virtual/issues/4). 

**OS:** Manjaro KDE Linux
**Python version:** 3.8

```
$ pip install -U pip
$ pip install titiler["server"]

# Or from sources
$ git clone https://github.com/developmentseed/titiler.git
$ cd titiler && pip install -e .["server"]
```
![](https://i.imgur.com/QSyyc7O.png)

**Comment by vincentsarago** [comment]
I think the problem comes from rio-color, which isn't really supported in conda. 
To fix this issue you need to first install numpy (with --no-binary) option 
@kylebarron has been pushing for this to be fixed but we are still waiting for the owner to move forward.
I'll update the docs 🙏

**Comment by kylebarron** [comment]
The issue here is that `rio-color` doesn't provide wheels ([yet](https://github.com/mapbox/rio-color/pull/66)), which means that to install `rio-color` all installations must build from source. Since it links to Numpy, it needs Numpy's headers while compiling.

This is a bit of a chicken-and-egg issue because if we define `numpy` as an installation requirement, it won't get installed before `rio-color` is attempted to be installed.

So I believe the correct solution here is to define `numpy` as a build dependency (or really `rio-color` should do that, I might make a PR) so that it's installed before `rio-color` is attempted to be installed.

**Comment by giswqs** [comment]
@vincentsarago @kylebarron Thank you both for clarifying this. Once this installation issue is fixed, I will add titlier as a dependency for [geemap](https://github.com/giswqs/geemap). Thank you for developing this awesome package!

**Comment by kylebarron** [comment]
@giswqs Note that it might be simplest for you to use [`rio-tiler`](https://github.com/cogeotiff/rio-tiler) directly.

```py
from rio_tiler import COGReader

with COGReader("my-tif.tif") as cog:
    # Read data for a mercator tile
    img: ImageData = cog.tile(tile_x, tile_y, tile_zoom, tilesize=256)
    assert img.data
    assert img.mask
```

Using a full server might be overkill for your needs.

Also note that looking at `geemap`'s dependencies, it looks like it doesn't currently require `GDAL`. Depending on `rio-tiler` -> `rasterio` -> `GDAL` will make installations larger and more complex.

**Comment by vincentsarago** [comment]
closed by https://github.com/developmentseed/titiler/commit/c7f8bfe978eae6e86eae10a29d879cd581274275

This is a temporary fix, and should be fixed fully when rio-color 1.0.0 comes out!

**Comment by kylebarron** [comment]
When rio-color 1.0.1 comes out
